pkg/querier/metadata_handler.go

```diff
@@ -56,8 +83,20 @@ func NewMetadataHandler(m MetadataSupplier) http.Handler {
 		// Put all the elements of the pseudo-set into a map of slices for marshalling.
 		metrics := map[string][]metricMetadata{}
 		for _, m := range resp {
+			// We enforce this both here and in the ingesters. Doing it in the ingesters is
+			// more efficient as it is earlier in the process, but since that one is per user,
+			// we still need to do it here after all the results are merged.
```

Comment:
I don't fully understand this. Given we enforce it in ingester, how can we receive metadata for a metric different than the requested one?

Reply:
That's more of a general comment on why we implement the logic for the request parameters there in addition to the ingesters, and applies more to the limits. You're right that it shouldn't matter for the metric, so I removed that one in https://github.com/grafana/mimir/pull/6064